webhooks-extension/docs/InstallPrereqs.md

```diff
@@ -0,0 +1,88 @@
+ 
+       - Clone this repository: `git clone https://github.com/tektoncd/experimental.git`
+       - Run the istio installation script: `./scripts/install_istio.sh <version>` 
+         
+         **Note:** requires Helm to be installed and recommend using 1.1.7 as `<version>`
+   
+   - **For a custom quickstart install:**
+ 
+       - Install as per instructions in the knative docs https://knative.dev/docs/install/installing-istio/ _(Istio version 1.1.7 is recommended)_
+
+   - **For official istio install steps:**
+
+       - Install as per instructions in the istio documentation https://istio.io/docs/setup/kubernetes/install/
+
+4. Install [Knative Eventing, Eventing Sources & Serving](https://knative.dev/docs/install/index.html)
+
+    The initial webhokks-extension implementation utilises Knative Eventing but there is discussion and work in [Tekton Pipelines](https://github.com/tektoncd/pipeline) to minimize necessary componentry in future.
+
+   - **For a quickstart install:**
+   
+       - Clone this repository: `git clone https://github.com/tektoncd/experimental.git`
+       - Run the knative installation script: `./scripts/install_knative.sh <version>` 
+         
+         **Note:** recommend using 0.6.0 as `<version>`
+
+   - **For official knative install steps:**
+   
+       - Install as per instructions in the knative documentation https://knative.dev/docs/install/index.html
+
+5.  Configure knative serving (see below)  
+
+
+## Domain setup for knative serving
+
+After installing the prereqs, set your own domain and selectors following the [configuring Knative Serving docs](https://github.com/knative/serving/blob/master/install/CONFIG.md) which outlines setting up routes in the `config-domain` ConfigMap in the `knative-serving` namespace.
+
+**Example using the cluster master node's DNS name**
+
+  Patch the ConfigMap, substituting ${cluster_master_node_dns_name} with the relevant DNS name:
+
+    ```
+    kubectl patch configmap config-domain --namespace knative-serving --type='json' \
+      --patch '[{"op": "add", "path": "/data/'"${cluster_master_node_dns_name}"'", "value": ""}]'
+    ```
+
+**Example using IP**
+
+  Retrieve your IP:
+
+    `ip=$(ifconfig | grep netmask | sed -n 2p | cut -d ' ' -f2)`
+
+  And patch it to the ConfigMap:
+
+    ```
+    kubectl patch configmap config-domain --namespace knative-serving --type='json' \
+      --patch '[{"op": "add", "path": "/data/'"${ip}.nip.io"'", "value": ""}]'
+    ```  
+
+This setup is important, as it will be used on the webhook's payload URL - Remember that your source code repository must be able to reach your cluster, or your webhooks will never be received (mentioning "Service Timeout" errors).
+
+If using GitHub enterprise you may simply be able to use your system IP address, if using the GitHub.com, your settings will need to be such that GitHub.com can route to the endpoint through any firewall therefore you might need to speak to your network administrator.
```

Comment:
if using GitHub.com, no need for `the GitHub.com`

`might need to speak to your network administrator`
😆 can we please reword...

Reply:
done